### PR TITLE
add some more nrfx clock apis

### PR DESCRIPTION
**Pull request by auroraslb**
Added some more nrfx clock APIs.

### CODE REVIEW

**Comment by rugeGerritsen** [comment]
@aescolar . Can you take a look at this one?

**Comment by aescolar** [comment]
@auroraslb Please note that you need to sign-off the commit

File: src/nrfx/hal/nrf_clock.c
Comment (aescolar) [inline src/nrfx/hal/nrf_clock.c]: Note that this is already happening in `nrf_clock_TASKS_HFCLKSTART()` (called from `nrf_clock_reqw_sideeffects_TASKS_HFCLKSTART()` just below) so doing it here is unnecessary.

A bit of background: the pattern in this models is that the HAL files only do what you expect the real HAL to do, say set a register (which in real life automatically triggers a sideeffect in the HW, but in the models the sideeffect needs to be emulated by calling a function in the HW models from the HAL replacement. In this case this is to set `TASKS_HFCLKSTART` and calling  `nrf_clock_reqw_sideeffects_TASKS_HFCLKSTART()` ).

But all HW behaviour is modeled in the HW_models files. So toggling status registers like HFCLKRUN would be done in the equivalent HW model functions.

File: src/nrfx/hal/nrf_clock.c
Comment (aescolar) [inline src/nrfx/hal/nrf_clock.c]: Same as above but in `nrf_clock_TASKS_LFCLKSTART()`

File: src/nrfx/hal/nrf_clock.c
Comment (aescolar) [inline src/nrfx/hal/nrf_clock.c]: Following what is described above, you would need to create the sideeffect function in NRF_clock.c and clear that LFCLKRUN there.
In any case, note that the models did not expect anywhere the LF clock would ever be turned off. Meaning for example the RTC model assumes its LFCLK input keeps running forever.
What is the use case in which this needs to be modeled?

File: src/nrfx/hal/nrf_clock.c
Comment (aescolar) [inline src/nrfx/hal/nrf_clock.c]: Without sideeffects these 3 ones are not going to do anything.
Note that setting the TASK register is a very phony thing in the models, which is immediately cleared by the HW model.

File: src/nrfx/hal/nrf_clock.c
Comment (aescolar) [inline src/nrfx/hal/nrf_clock.c]: Maybe better place this in the header, just like in the real HAL. The hope is to minimize the differences between this replacement HAL and the real Nordic HAL. So for functions which can be the same, better for them to be the same and be placed in the same files as in the Nordic HAL.
(You should only need to add `static inline` before `bool`. And you should not need to do the change of replacing `p_reg->`  with `NRF_CLOCK_regs.`)

Same thing about nrf_clock_lf_srccopy_get and nrf_clock_hf_src_get below.

(In that regard, I should have placed myself `nrf_clock_int_enable_check` in the header...)

**Review by aescolar — COMMENTED** [review]
Thanks @auroraslb ,
See a few requests below. Please note that to be able to accept your changes we will also need you to sign them off (`git commit -amend -s`)  https://babblesim.github.io/contribution_guidelines.html

File: src/nrfx/hal/nrf_clock.c
Comment (auroraslb) [reply src/nrfx/hal/nrf_clock.c]: In our use case we are stopping and starting the clock to reconfigure a different source. We therefore do not need to model this. Is it OK to leave it as it is?

File: src/nrfx/hal/nrf_clock.c
Comment (auroraslb) [reply src/nrfx/hal/nrf_clock.c]: Should I just remove writing to tasks?

File: src/nrfx/hal/nrf_clock.c
Comment (aescolar) [reply src/nrfx/hal/nrf_clock.c]: I'd ask you to create a new `nrf_clock_reqw_sideeffects_TASKS_LFCLKSTOP` next to nrf_clock_reqw_sideeffects_TASKS_LFCLKSTART() and set that register to 0 there, with a comment in the code describing that no effect of turning the clock off is actually modeled

File: src/nrfx/hal/nrf_clock.c
Comment (aescolar) [reply src/nrfx/hal/nrf_clock.c]: Depends on what is it that you are trying to do.
Why did you add these 3 conditions? (NRF_CLOCK_TASK_CAL, NRF_CLOCK_TASK_CTSTART and NRF_CLOCK_TASK_CTSTOP)
if they have no purpose, then better to not have the conditions, and have the error below be thrown to users so they know nothing happens.
If you cannot have that error, please create a stub'ed function in the NRF_CLOCK.c HW model file for each of these, which clears back the task register, and maybe prints a warning

File: src/HW_models/NRF_CLOCK.c
Comment (aescolar) [inline src/HW_models/NRF_CLOCK.c]: shouldnt this be LFCLKSTOP?

File: src/HW_models/NRF_CLOCK.h
Comment (wopu-ot) [inline src/HW_models/NRF_CLOCK.h]: I think this should be called `nrf_clock_reqw_sideeffects_TASKS_LFCLKSTOP` to be in line with the other functions.

Comment (aescolar) [reply src/HW_models/NRF_CLOCK.h]: I agree with @wopu-ot (and sorry for the typo in my original comment)

File: src/nrfx/hal/nrf_clock.c
Comment (aescolar) [reply src/nrfx/hal/nrf_clock.c]: That warning is probably good enough. (Though I tend to prefer things to clearly blob up when a non implemented feature is used, unless it is a known missing feature)

File: src/nrfx/hal/nrf_clock.h
Comment (aescolar) [inline src/nrfx/hal/nrf_clock.h]: Please add these 2 after nrf_clock_is_running() so they are ordered like in the original HAL file (so it is much easier to find differences)

**Review by wopu-ot — APPROVED** [review]
LGTM

**Review by wopu-ot — CHANGES_REQUESTED** [review]
As observed by @aescolar, the call to `nrf_clock_reqw_sideeffects_TASK_LFCLKSTOP` is lacking the renaming.

**Comment by rugeGerritsen** [comment]
> As observed by @aescolar, the call to `nrf_clock_reqw_sideeffects_TASK_LFCLKSTOP` is lacking the renaming.

@wopu-ot . I might be wrong. But it looks like the function is now correctly named `nrf_clock_reqw_sideeffects_TASKS_LFCLKSTOP`

File: src/nrfx/hal/nrf_clock.c
Comment (aescolar) [inline src/nrfx/hal/nrf_clock.c]: here :)

Comment (rugeGerritsen) [reply src/nrfx/hal/nrf_clock.c]: ah :)

**Review by aescolar — APPROVED** [review]

**Comment by aescolar** [comment]
Hi @auroraslb , after I merged another PR there is a conflict in this one. I took the liberty of resolving the conflict and doing 2 minor fixes in your branch here :
github.com:aescolar/ext_NRF52_hw_models.git
https://github.com/aescolar/ext_NRF52_hw_models/tree/add_more_clock_apis
If you would pick that into this PR we should be able to merge it.


To replace your local branch with that one you can do
```
git remote add aescolar https://github.com/aescolar/ext_NRF52_hw_models
git fetch aescolar add_more_clock_apis
git checkout add_more_clock_apis
git reset --hard aescolar/add_more_clock_apis
```

**Review by aescolar — APPROVED** [review]

**Review by wopu-ot — APPROVED** [review]

**Pull request merged**
